src/test/java/uk/ac/ebi/ampt2d/commons/accession/utils/ExponentialBackOffTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.utils;
+
+import org.junit.Test;
+import uk.ac.ebi.ampt2d.commons.accession.utils.exceptions.ExponentialBackOffMaxRetriesRuntimeException;
+
+import java.util.function.Supplier;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ExponentialBackOffTest {
+
+    private class ExtendedRunnable implements Runnable, Supplier<String> {
+
+        private int numRun = 0;
+
+        @Override
+        public void run() {
+            numRun++;
+            throw new RuntimeException();
+        }
+
+        @Override
+        public String get() {
+            numRun++;
+            throw new RuntimeException();
+        }
+    }
+
+    @Test(expected = ExponentialBackOffMaxRetriesRuntimeException.class)
+    public void testRunnableDefaultTotalAttempts() {
```

Comment:
Is there a way to make this test faster? If not, I would "ignore it" by default. (and testFunctionDefaultTotalAttempts too)